addons/knobs/src/components/types/Select.tsx

```diff
@@ -4,13 +4,13 @@ import PropTypes from 'prop-types';
 import { Form } from '@storybook/components';
 import { KnobControlConfig, KnobControlProps } from './types';
 
-export type SelectTypeKnobValue = string | number | null | undefined | PropertyKey[];
+export type SelectTypeKnobValue = string | number | boolean | null | undefined | PropertyKey[];
 
 export type SelectTypeOptionsProp<T extends SelectTypeKnobValue = SelectTypeKnobValue> =
   | Record<PropertyKey, T>
   | Record<Extract<T, PropertyKey>, T[keyof T]>
-  | Extract<T, PropertyKey>[]
-  | readonly Extract<T, PropertyKey>[];
+  | T[]
+  | readonly T[];
```

Comment:
@gaetanmaisse I believe this was typed originally with `PropertyKey` out of caution given that option arrays get converted into objects by coercing the values into keys:

```
[undefined, null, false]  =>  { 'undefined': undefined, 'null': null, 'false': false }
```

However, in practice getting yourself into trouble because of this may be kind of a corner case, so this should be fine. The select control itself does equality checks against the values, so everything should work as expected.